packages/esm-patient-immunizations-app/src/immunizations/immunizations-form.component.tsx

```diff
@@ -1,91 +1,222 @@
+                titleText={t('sequence', 'Sequence')}
+                items={sequencies?.map((sequence) => sequence.sequenceNumber) || []}
+                itemToString={(item) => sequencies.find((s) => s.sequenceNumber === item)?.sequenceLabel}
+                onChange={(val) => onChange(parseInt(val.selectedItem || 0))}
+                selectedItem={value}
+              />
+            ) : (
+              <NumberInput
+                id="doseNumber"
+                label={t('doseNumber', 'Dose number within series')}
+                min={0}
+                onChange={(event) => onChange(parseInt(event.target.value || 0))}
+                value={value}
+                hideSteppers={true}
+              />
+            )
+          }
+        />
+      </div>
+    );
+  }, [immunizationsConfig, t, control, vaccineUuid]);
+
+  const onSubmit = React.useCallback(
+    (data: ImmunizationFormInputData) => {
+      setIsSubmitting(true);
+      const {
+        vaccineUuid,
+        vaccinationDate,
+        doseNumber,
+        expirationDate,
+        lotNumber,
+        manufacturer,
+        timeFormat,
+        vaccinationTime,
+      } = data;
+      const abortController = new AbortController();
+
+      const [hours, minutes] = convertTime12to24(vaccinationTime, timeFormat);
 
       const immunization: ImmunizationFormData = {
         patientUuid,
-        ...formState,
+        immunizationId: immunizationToEditMeta?.immunizationObsUuid,
+        vaccineName: immunizationsConceptSet.answers.find((answer) => answer.uuid === vaccineUuid).display,
+        vaccineUuid: vaccineUuid,
+        vaccinationDate: toDateObjectStrict(
+          toOmrsIsoString(
+            new Date(
+              dayjs(vaccinationDate).year(),
+              dayjs(vaccinationDate).month(),
+              dayjs(vaccinationDate).date(),
+              hours,
+              minutes,
+            ),
+          ),
+        ),
+        doseNumber,
+        expirationDate,
+        lotNumber,
+        manufacturer,
       };
 
-      const abortController = new AbortController();
-
       savePatientImmunization(
-        mapToFHIRImmunizationResource(immunization, currentVisitUuid, currentLocationUuid, currentProviderUuid),
-        patientUuid,
-        formState.immunizationObsUuid,
+        mapToFHIRImmunizationResource(
+          immunization,
+          immunizationToEditMeta?.visitUuid || currentVisit?.uuid,
+          currentUser?.sessionLocation?.uuid,
+          currentUser?.currentProvider?.uuid,
+        ),
+        immunizationToEditMeta?.immunizationObsUuid,
         abortController,
       ).then(
         (response) => {
-          response.status === 201 && closeWorkspace();
+          response.ok && closeWorkspace();
           showSnackbar({
             kind: 'success',
             title: t('vaccinationSaved', 'Vaccination saved successfully'),
             isLowContrast: true,
           });
```

Comment:
```suggestion
          if (response.ok) {
            closeWorkspace();
            showSnackbar({
              kind: 'success',
              title: t('vaccinationSaved', 'Vaccination saved successfully'),
              isLowContrast: true,
            });
         }
```

Reply:
Also do we need the `response.ok` check at all?

Reply:
This is the point where we should call the bound `mutate` function to trigger a revalidation each time a new immunization is recorded.

Reply:
>Also do we need the response.ok check at all?

The existence of the error block renders it insignificant. Just added it to be consistent with the existing patterns in the codebase.

Reply:
I mean, we should also retire it there because we [already do the check in `openmrsFetch()`](https://github.com/openmrs/openmrs-esm-core/blob/43a20b2cc5673071ed9c816c20a93682cb9e977e/packages/framework/esm-api/src/openmrs-fetch.ts#L153). `openmrsFetch()` will throw if `response.ok` is false.